### PR TITLE
SDI-493 Alert if size of index does not match the number of prisoners…

### PR DESCRIPTION
**Pull request by steverendell**
… in NOMIS

performance and accuracy improvement

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/resource/GlobalSearchResource.kt [hunk 1]
```diff
@@ -76,8 +76,8 @@ class GlobalSearchResource(
       PageRequest.of(0, 10)
     )
     val mid = System.currentTimeMillis()
-    val totalNomisNumber = getTotalNomisNumber()
     val totalIndexNumber = prisonerIndexService.countIndex(indexStatusService.getCurrentIndex().currentIndex)
+    val totalNomisNumber = getTotalNomisNumber()
```
Comment (steverendell) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/resource/GlobalSearchResource.kt:80]: This should fix the problem of the index sometimes being > nomis. I think that prisoners can be created during the slow nomis query, so getting the index first means it is more likely to equal the nomis count (which will be that at the _start_ of the nomis query execution)

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/GlobalSearchService.kt [hunk 2]
```diff
@@ -141,8 +144,7 @@ class GlobalSearchService(
   private fun setupIndexSearch(scroll: Scroll): SearchResponse {
 
     val searchSourceBuilder = SearchSourceBuilder().apply {
-      docValueField("prisonerNumber")
-      sort("prisonerNumber")
+      fetchSource(FetchSourceContext.DO_NOT_FETCH_SOURCE)
```
Comment (steverendell) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/GlobalSearchService.kt:147]: Ive realised that prisonerNumber is actually also the document id, so no need to get any fields - may be much faster 🤞 . Does mean a manual sort is needed however.

Comment (mikehalmamoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/GlobalSearchService.kt:147]: If you're getting the whole index up front maybe you could do a sort and *then* use Kotlin's minus to get the diffs? I love minus as you can tell 😄

**Review by lbennett-moj — DISMISSED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/GlobalSearchService.kt (hunk 2 above)
Comment (steverendell) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/GlobalSearchService.kt:147]: aww ... i cant let go of my lovely code ...... go on then! Ill do a minus

**Review by lbennett-moj — APPROVED** [review]
LGTM

**Pull request merged**